// @tailus/themer-button/dist/themes/default/variants/soft-variant/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.trailingIconSizes = exports.leadingIconSizes = exports.iconSizes = exports.getSoftIconOnlyButtonSizes = exports.getSoftButtonSizes = exports.getSoftIconButtonSizes = void 0;
var index_1 = require("./../index");
var colors_1 = require("./colors");
var softBase = {
    layout: "group flex items-center rounded-[--btn-border-radius] disabled:border *:select-none [&>*:not(.sr-only)]:relative *:disabled:opacity-20",
    appearances: {
        light: "disabled:text-gray-950 disabled:border-gray-200 disabled:bg-gray-100",
        dark: "dark:disabled:border-gray-800/50 disabled:dark:bg-gray-900 dark:*:disabled:!text-white",
        both: "disabled:text-gray-950 disabled:border-gray-200 disabled:bg-gray-100 dark:disabled:border-gray-800/50 disabled:dark:bg-gray-900 dark:*:disabled:!text-white",
    },
};
var getSoftButtonSizes = function (color, base, sizes) {
    if (base === void 0) { base = softBase; }
    if (sizes === void 0) { sizes = index_1.buttonSizes; }
    return (0, index_1.getButtonSizes)(color, base, colors_1.default, sizes);
};
exports.getSoftButtonSizes = getSoftButtonSizes;
var getSoftIconButtonSizes = function (color, sizes) {
    if (sizes === void 0) { sizes = index_1.buttonSizes; }
    return (0, index_1.getIconButtonSizes)(color, softBase, colors_1.default, sizes);
};
exports.getSoftIconButtonSizes = getSoftIconButtonSizes;
var getSoftIconOnlyButtonSizes = function (color) {
    return getSoftButtonSizes(color, softBase, index_1.iconOnlyButtonSizes);
};
exports.getSoftIconOnlyButtonSizes = getSoftIconOnlyButtonSizes;
var index_2 = require("./../index");
Object.defineProperty(exports, "iconSizes", { enumerable: true, get: function () { return index_2.iconSizes; } });
Object.defineProperty(exports, "leadingIconSizes", { enumerable: true, get: function () { return index_2.leadingIconSizes; } });
Object.defineProperty(exports, "trailingIconSizes", { enumerable: true, get: function () { return index_2.trailingIconSizes; } });
