// @tailus/themer-button/dist/themes/default/variants/ghost-variant/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.trailingIconSizes = exports.leadingIconSizes = exports.iconSizes = exports.getGhostIconOnlyButtonSizes = exports.getGhostButtonSizes = exports.getGhostIconButtonSizes = void 0;
var index_1 = require("./../index");
var colors_1 = require("./colors");
var ghostBase = {
    layout: "group flex items-center rounded-[--btn-border-radius] disabled:border *:select-none [&>*:not(.sr-only)]:relative *:disabled:opacity-20",
    appearances: {
        light: "*:disabled:text-gray-950 disabled:border-gray-200 disabled:bg-gray-100",
        dark: "dark:disabled:border dark:disabled:border-gray-800 disabled:dark:bg-gray-900 dark:*:disabled:!text-white",
        both: "*:disabled:text-gray-950 disabled:border-gray-200 disabled:bg-gray-100 dark:disabled:border dark:disabled:border-gray-800 disabled:dark:bg-gray-900 dark:*:disabled:!text-white",
    },
};
var getGhostButtonSizes = function (color, base, sizes) {
    if (base === void 0) { base = ghostBase; }
    if (sizes === void 0) { sizes = index_1.buttonSizes; }
    return (0, index_1.getButtonSizes)(color, base, colors_1.default, sizes);
};
exports.getGhostButtonSizes = getGhostButtonSizes;
var getGhostIconButtonSizes = function (color, base, sizes) {
    if (base === void 0) { base = ghostBase; }
    if (sizes === void 0) { sizes = index_1.buttonSizes; }
    return (0, index_1.getIconButtonSizes)(color, base, colors_1.default, sizes);
};
exports.getGhostIconButtonSizes = getGhostIconButtonSizes;
var getGhostIconOnlyButtonSizes = function (color, base) {
    if (base === void 0) { base = ghostBase; }
    return getGhostButtonSizes(color, base, index_1.iconOnlyButtonSizes);
};
exports.getGhostIconOnlyButtonSizes = getGhostIconOnlyButtonSizes;
var index_2 = require("./../index");
Object.defineProperty(exports, "iconSizes", { enumerable: true, get: function () { return index_2.iconSizes; } });
Object.defineProperty(exports, "leadingIconSizes", { enumerable: true, get: function () { return index_2.leadingIconSizes; } });
Object.defineProperty(exports, "trailingIconSizes", { enumerable: true, get: function () { return index_2.trailingIconSizes; } });
