// @tailus/themer-card/dist/themes/default/elevatedGradientVariant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.elevatedGradientVariant = void 0;
var card_config_1 = require("../../card.config");
var base = "p-px rounded-[--card-border-radius] bg-gradient-to-b to-50% card-shadow";
var appearances = {
    light: "from-[--ui-light-border-color] to-[--card-light-bg]",
    dark: "from-[--ui-dark-border-color] to-[--card-dark-bg]",
    both: "from-[--ui-light-border-color] to-[--card-light-bg] dark:from-[--ui-dark-border-color] dark:to-[--card-dark-bg]",
};
var innnerCard = {
    base: "p-[calc(var(--card-padding)-1px)] rounded-[calc(var(--card-border-radius)-1px)]",
    appearances: {
        light: "bg-white",
        dark: "bg-[--card-dark-bg]",
        both: "bg-white dark:bg-[--card-dark-bg]",
    },
};
var elevatedGradientVariant = {
    outer: base + " " + appearances[card_config_1.default.appearance],
    inner: innnerCard.base + " " + innnerCard.appearances[card_config_1.default.appearance],
};
exports.elevatedGradientVariant = elevatedGradientVariant;
exports.default = elevatedGradientVariant;
