// @tailus/themer-progress/dist/themes/default/base.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.timer = exports.message = exports.label = exports.indicator = exports.baseIndicator = exports.header = exports.root = void 0;
var colors_1 = require("./colors");
var progress_config_1 = require("./../../progress.config");
var baseRoot = {
    base: "relative h-1.5 overflow-hidden rounded-full",
    appearances: {
        light: "bg-gray-950/10",
        dark: "bg-white/10",
        both: "bg-gray-950/10 dark:bg-white/10",
    },
};
var baseIndicator = "ease-[cubic-bezier(0.65, 0, 0.35, 1)] h-full w-full transition-transform duration-[660ms]";
exports.baseIndicator = baseIndicator;
var baseLabel = {
    base: "font-medium text-sm",
    appearances: {
        light: "text-gray-800",
        dark: "text-white",
        both: "text-gray-800 dark:text-white",
    },
};
var baseMessage = {
    base: "mt-2 block text-sm",
    appearances: {
        light: "text-gray-600",
        dark: "text-gray-400",
        both: "text-gray-600 dark:text-gray-400",
    },
};
var baseTimer = {
    base: "text-xs tracking-wide gap-1 items-center flex",
    appearances: {
        light: "text-gray-600",
        dark: "text-gray-400",
        both: "text-gray-600 dark:text-gray-400",
    },
};
var indicator = {
    primary: (0, colors_1.getIndicatorColor)("primary"),
    secondary: (0, colors_1.getIndicatorColor)("secondary"),
    success: (0, colors_1.getIndicatorColor)("success"),
    info: (0, colors_1.getIndicatorColor)("info"),
    warning: (0, colors_1.getIndicatorColor)("warning"),
    danger: (0, colors_1.getIndicatorColor)("danger"),
    gray: (0, colors_1.getIndicatorColor)("gray"),
    neutral: (0, colors_1.getIndicatorColor)("neutral"),
    gradient: (0, colors_1.getIndicatorColor)("gradient"),
};
exports.indicator = indicator;
var root = baseRoot.base + " " + baseRoot.appearances[progress_config_1.default.appearance];
exports.root = root;
var header = "items-center mb-4 flex justify-between";
exports.header = header;
var label = baseLabel.base + " " + baseLabel.appearances[progress_config_1.default.appearance];
exports.label = label;
var message = baseMessage.base + " " + baseMessage.appearances[progress_config_1.default.appearance];
exports.message = message;
var timer = baseTimer.base + " " + baseTimer.appearances[progress_config_1.default.appearance];
exports.timer = timer;
